packages/entrykit/src/onboarding/deposit/DepositForm.tsx

```diff
@@ -0,0 +1,148 @@
+import { useEffect, useRef } from "react";
+import { useAccount, useBalance, useWatchBlockNumber } from "wagmi";
+import { useBalance as useQuarryBalance } from "../quarry/useBalance";
+import { ChainSelect } from "./ChainSelect";
+import { AmountInput } from "./AmountInput";
+import { twMerge } from "tailwind-merge";
+import { PendingIcon } from "../../icons/PendingIcon";
+import { formatGas } from "./formatGas";
+import { SubmitButton } from "./SubmitButton";
+import { useIsMounted } from "usehooks-ts";
+import { WarningIcon } from "../../icons/WarningIcon";
+import { Balance } from "./Balance";
+import { useShowQueryError } from "../../errors/useShowQueryError";
+import { Chain } from "viem";
+
+export const DEFAULT_DEPOSIT_AMOUNT = 0.005;
+
+export type Props = {
+  sourceChain: Chain;
+  setSourceChainId: (chainId: number) => void;
+  amount: bigint | undefined;
+  setAmount: (amount: bigint | undefined) => void;
+  estimatedFee: {
+    fee?: bigint | undefined;
+    isLoading?: boolean | undefined;
+    error: Error | undefined;
+  };
+  estimatedTime: string;
+  onSubmit: () => Promise<void>;
+  submitButton: React.ReactNode;
+};
+
+export function DepositForm({
+  sourceChain,
+  setSourceChainId,
+  amount,
+  setAmount,
+  estimatedFee,
+  estimatedTime,
+  onSubmit,
+  submitButton,
+}: Props) {
+  const amountInputRef = useRef<HTMLInputElement | null>(null);
+  const isMounted = useIsMounted();
+
+  const { address: userAddress, chainId: userChainId } = useAccount();
+  const balance = useShowQueryError(useBalance({ chainId: sourceChain.id, address: userAddress }));
+  const quarryBalance = useShowQueryError(useQuarryBalance(userAddress));
+  useWatchBlockNumber({
+    onBlockNumber: () => {
+      balance.refetch();
+      quarryBalance.refetch();
+    },
+  });
+
+  const minimumBalance = amount != null ? amount + (estimatedFee?.fee ?? 0n) : undefined;
+  const hasMinimumBalance = balance.data != null ? balance.data.value > (minimumBalance ?? 0n) : undefined;
+
+  // Re-focus input if chain ID changes (otherwise the chain select is still in focus)
+  useEffect(() => {
+    amountInputRef.current?.focus();
+  }, [userChainId]);
+
+  useEffect(() => {
+    if (balance.error) {
+      console.error("Failed to get balance for", userAddress, "on", sourceChain.id, balance.error);
+    }
+  }, [balance.error, sourceChain.id, userAddress]);
+
+  useEffect(() => {
+    if (estimatedFee.error) {
+      console.error("Failed to estimate fee, deposit from", sourceChain.id, estimatedFee.error);
+    }
+  }, [estimatedFee.error, sourceChain.id, userAddress]);
```

Comment:
I think both of these errors should already be handled by `useShowQueryError` wrapper